internal/exp/metrics/staleness/staleness.go

```diff
@@ -102,3 +102,33 @@ func (s *Staleness[T]) Evict() (identity.Stream, bool) {
 func (s *Staleness[T]) Clear() {
 	s.items.Clear()
 }
+
+type Tracker struct {
```

Comment:
Interesting. So with Tracker you no longer own the data, but instead use the id as a handle. And the processor owns the data. I can get behind that